core/divesoft.cpp

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: GPL-2.0
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "errorhelper.h"
+#include "subsurface-string.h"
+#include "gettext.h"
+#include "dive.h"
+#include "divelist.h"
+#include "divelog.h"
+#include "extradata.h"
+#include "format.h"
+#include "libdivecomputer.h"
+
+int divesoft_import(const std::unique_ptr<std::vector<unsigned char>> &buffer, struct divelog *log)
+{
+	int model = 0;
+	if (strncmp((char *)(buffer->data() + 52), "7026", 4) == 0)
+		model = 10;
```

Comment:
Please add a comment saying that its is Liberty, and the same for the 19 bellow for Freedom.

Reply:
I actually think this whole model bit here is unnecessary, because all the divesoft models use the same parser, and the log will get the right model anyway from the DEVINFO event.

Reply:
> I actually think this whole model bit here is unnecessary, because all the divesoft models use the same parser, and the log will get the right model anyway from the DEVINFO event.

libdivecomputer does not allow us to _not_ specify a model here - and supplying what is potentially the incorrect model seems to be a questionable approach.
Even if the Divesoft parser does not depend on the model information _at the moment_, there are no guarantees that this will not change in a future version, and introduce incorrect parsing when libdivecomputer is updated.
Also, note that the `DEVINFO` event is emitted by the Divesoft device handler, and not by the parser, so this doesn't happen when we are reading log files.

Reply:
You just need to pass a valid value, no reason to pass a specific value. I'd prefer if we'd skip the guessing game. The only relevant thing is to pass the correct dc_family_t.

Also, this variant of guessing game based on serial is only valid for the new log file format, and not for the old one.

That whole thing is kinda why we'd drop the old parser and accept the loss of functionality to user the libdivecomputer one.

The actual model is available in the cfg-sample, but that would involve more parsing of the file, which was exactly what we didn't want to do.

Also, just looking at the amount of information that's missing in the libdivecomputer parser makes me question the chose to drop our current parser.

Reply:
> Also, just looking at the amount of information that's missing in the libdivecomputer parser makes me question the chose to drop our current parser.

I don't mind either way, if somebody wants to put in the effort to add support for the models currently sold by Divesoft to our standalone logfile parser then that's ok with me.

[This problem](https://groups.google.com/g/subsurface-divelog/c/FWh8gwfE444) has been reported to us two months ago, and nothing has happened on our side, so it does not look like there is motivation to get our own implementation of this parser updated, and likely also not to put in the effort to incorporate any future changes in the format. During the same two months, libdivecomputer has updated their parser to support the current models, so we can get this essentially 'for free', together with any future updates.

In general, seeing that we are struggling to find the people willing to do the work to keep this project maintained, it seems to me that keeping two parsers for the same format maintained is a wasteful approach that doesn't fit with where we are at. If we are not happy with the scope of functionality in the parser included in libdivecomputer then it might be a better use of time to add the functionality that we would like to have to it, instead of maintaining an additional implementation of the same parser.

Reply:
I couldn't agree more.
Duplicative work is the last thing we want to do.